src/Func.cpp

```diff
@@ -1090,6 +1090,36 @@ void Stage::split(const string &old, const string &outer, const string &inner, c
             << "Use TailStrategy::GuardWithIf instead.";
     }
 
+    bool predicate_loads_ok = !exact;
```

Comment:
I think PredicateLoads might actually be legal when exact is true. Let's discuss offline.

Reply:
No, exact is when you're splitting an rvar, which may scatter, so it's not legal to use PredicateLoads because it may scatter garbage values.

Reply:
which means we skip this line and the check?

Reply:
It depends which error message we want to trigger first. I guess we want the second one to trigger, because the first one says "you can't use PredicateLoads on this Var in this way", and the second says "you can't use PredicateLoads on this Var at all". So keep this line and the check.

Reply:
reflected, I think the PR is now ready to land